src/cursor/cur_file.c

```diff
@@ -937,6 +937,34 @@ __curfile_setup_checkpoint(WT_CURSOR_BTREE *cbt, const char *cfg[], WT_DATA_HAND
     return (ret);
 }
 
+/*
+ * __curfile_bound --
+ *     WT_CURSOR->bound implementation for file cursors.
+ */
+static int
+__curfile_bound(WT_CURSOR *cursor, const char *config)
+{
+    WT_COLLATOR *btree_collator;
+    WT_CURSOR_BTREE *cbt;
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    cbt = (WT_CURSOR_BTREE *)cursor;
+
+    CURSOR_API_CALL(cursor, session, ret, bound, NULL);
+
+    if (CUR2BT(cursor)->type == BTREE_COL_FIX)
+        WT_ERR_MSG(session, EINVAL, "setting bounds is not compatible with fixed column store");
+    /* TODO: limit this to cases where a bound is being set, not cleared */
+    if (WT_CURSOR_IS_POSITIONED(cbt))
```

Comment:
@ddanderson I could use your help with the configuration compilation code. I want to move this check that is btree specific up into this btree level interface, but it requires checking a configuration option.

Is there a pattern for compile the configuration options here and pass them down? I was thinking __wti_cursor_bound could take a `WT_CONF_API_TYPE(WT_CURSOR_bound)` (I don't think it's quite right, but hopefully you get the idea).

Is that feasible? Is there an example of doing that somewhere?

Reply:
@agorrod Yes, the type of the thing you want to pass is a `WT_CONF *`.  The closest example is `__wt_txn_config(WT_SESSION_IMPL *session, WT_CONF *conf)` which is called to do all the configuration checks for txn.

So yeah, one way is to separate __wti_cursor_bound so it just creates/initializes the `WT_CONF *conf` and then calls a second function __wti_cursor_bound_inner with the `conf` that does the all the rest of the bounds stuff.  And __curfile_bound would also create/initialize the `WT_CONF *conf` and calls __wt_cursor_bound_inner.  Maybe with better naming...

Reply:
Thanks Don - that was exactly what I was looking for! I've made that change, hopefully it'll pass testing again.